ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js

```diff
@@ -12,181 +12,300 @@
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
  **/
-const React = require('react')
-
+import React, { useState } from 'react'
 const { Radio, RadioItem } = require('../radio')
 const TextField = require('../text-field')
-
+import { getErrorComponent, validateGeo } from '../utils/validation'
 const { Units, Zone, Hemisphere, MinimumSpacing } = require('./common')
-
 const {
   DmsLatitude,
   DmsLongitude,
 } = require('../../component/location-new/geo-components/coordinates.js')
 const DirectionInput = require('../../component/location-new/geo-components/direction.js')
 const { Direction } = require('../../component/location-new/utils/dms-utils.js')
-import styled from 'styled-components'
 
-const ErrorBlock = styled.div`
-  width: 100%;
-  display: flex;
-  flex-direction: row;
-  justify-content: flex-start;
-  align-items: center;
-  background: ${({ theme }) => theme.negativeColor};
-`
-
-const WarningIcon = styled.span`
-  padding: ${({ theme }) => theme.minimumSpacing};
-`
-
-const PointRadiusLatLon = props => {
-  const { lat, lon, radius, radiusUnits, cursor } = props
+const PointRadiusLatLonDd = props => {
+  const { lat, lon, radius, radiusUnits, setState } = props
+  const [ddError, setDdError] = useState({
+    error: false,
+    message: '',
+    defaultValue: '',
+  })
+  const [radiusError, setRadiusError] = useState({ error: false, message: '' })
   return (
     <div>
       <TextField
         type="number"
         label="Latitude"
-        value={lat}
-        onChange={cursor('lat')}
-        onBlur={props.callback}
+        value={lat !== undefined ? String(lat) : lat}
+        onChange={value => {
+          const { error, message, defaultValue } = validateGeo('lat', value)
+          if (defaultValue) {
+            setDdError({ error, message, defaultValue })
+          }
+          defaultValue ? setState('lat', defaultValue) : setState('lat', value)
+        }}
+        onFocus={() => {
+          setDdError({ error: false, message: '', defaultValue: '' })
+        }}
+        onBlur={() => setDdError(validateGeo('lat', lat))}
         addon="°"
       />
       <TextField
         type="number"
         label="Longitude"
-        value={lon}
-        onChange={cursor('lon')}
+        value={lon !== undefined ? String(lon) : lon}
+        onChange={value => {
+          const { error, message, defaultValue } = validateGeo('lon', value)
+          if (defaultValue) {
+            setDdError({ error, message, defaultValue })
+          }
+          defaultValue ? setState('lon', defaultValue) : setState('lon', value)
+        }}
+        onBlur={() => setDdError(validateGeo('lon', lon))}
         addon="°"
       />
-      <Units value={radiusUnits} onChange={cursor('radiusUnits')}>
+      {getErrorComponent(ddError)}
+      <Units
+        value={radiusUnits}
+        onChange={value => setState('radiusUnits', value)}
+      >
         <TextField
           type="number"
           min="0"
```

Comment:
should this be here still?

Reply:
these are the units i.e. miles, kilometers, etc

Reply:
I'm pretty sure this is in the radius text field itself

Reply:
ohh sorry i was looking at the changed part, up to line 72